### ISSUE TITLE
Table: Incorrect Input Attribute Transformation

### ISSUE THREAD

**Issue by lothern** [primefaces/primeng#15253]
### Describe the bug

For two of the input properties on the table component there is an incorrect attribute transformation being applied.

The two inputs are:
```
/**
 * Map instance to keep the expanded rows where key of the map is the data key of the row.
 * @group Props
 */
@Input({ transform: booleanAttribute }) expandedRowKeys: { [s: string]: boolean } = {};
/**
 * Map instance to keep the rows being edited where key of the map is the data key of the row.
 * @group Props
 */
@Input({ transform: booleanAttribute }) editingRowKeys: { [s: string]: boolean } = {};
```

The type for each doesn't need a boolean attribute transformation as the type of the property is an object.

This causes a runtime error:
> ERROR Error: Cannot create property '1' on boolean 'true'

### Environment

Angular 17.3.3
PrimeNG 17.13.0

### Reproducer

https://stackblitz.com/edit/github-vj4cq8?file=src%2Fapp%2Fapp.component.html

### Angular version

17.3.3

### PrimeNG version

17.13.0

### Build / Runtime

Angular CLI App

### Language

TypeScript

### Node version (for AoT issues node --version)

18.18.2

### Steps to reproduce the behavior

In provided reproducer:

1. Click the row expansion button
2. Observe the error in the console

### Expected behavior

I expect to pass an object conforming to the type of these inputs and not have it transformed/interpreted as a boolean type.

**Comment by w0wka91** [comment]
Is there a quickfix for this issue? I am still not sure if this issue came through the angular or the primeng update.

**Comment by BGBRWR** [comment]
https://github.com/primefaces/primeng/pull/15243 fixes this.

**Comment by lothern** [comment]
> Is there a quickfix for this issue? I am still not sure if this issue came through the angular or the primeng update.

Afraid not. I reverted back to PrimeNG 17.11.0 until this could be resolved.

**Comment by lothern** [comment]
This issue is resolved in 17.14.1.

Thank you!
